core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3207,14 +3212,55 @@ private boolean syncMetadata(RpcContext rpcContext, LockedInodePath inodePath,
       }
     }
 
-    if (loadMetadata) {
+    // Update metadata for all the mount points
+    for (String mountPoint : pathsToLoad) {
+      AlluxioURI mountPointUri = new AlluxioURI(mountPoint);
       try {
-        loadMetadataAndJournal(rpcContext, inodePath, LoadMetadataOptions.defaults()
-            .setCreateAncestors(true).setLoadDescendantType(syncDescendantType));
-      } catch (Exception e) {
-        // This may be expected. For example, when creating a new file, the UFS file is not
-        // expected to exist.
-        LOG.debug("Failed to load metadata for path: {}", inodePath.getUri(), e);
+        if (PathUtils.hasPrefix(inodePath.getUri().getPath(), mountPointUri.getPath())) {
+          // one of the mountpoint is above the original inodePath, we start loading from the
+          // original inodePath. It is already locked. so we proceed to load metadata.
+          try {
+            loadMetadataAndJournal(rpcContext, inodePath, LoadMetadataOptions.defaults()
+                .setCreateAncestors(true).setLoadDescendantType(syncDescendantType));
+          } catch (Exception e) {
+            // This may be expected. For example, when creating a new file, the UFS file is not
+            // expected to exist.
+            LOG.debug("Failed to load metadata for path: {}", inodePath.getUri(), e);
+            return false;
```

Comment:
Why return early here? Shouldn't we continue processing the rest of the `pathsToLoad`?